Two Sums - Unsorted/Two Sums - Unsorted.js

```diff
@@ -0,0 +1,57 @@
+/*
+
+Given an array of integers nums and an integer target, return indices of the two numbers such that they add up to target.
+
+You may assume that each input would have exactly one solution, and you may not use the same element twice.
+
+You can return the answer in any order.
+
+ 
+
+Example 1:
+
+Input: nums = [2,7,11,15], target = 9
+Output: [0,1]
+Explanation: Because nums[0] + nums[1] == 9, we return [0, 1].
+Example 2:
+
+Input: nums = [3,2,4], target = 6
+Output: [1,2]
+Example 3:
+
+Input: nums = [3,3], target = 6
+Output: [0,1]
+ 
+
+Constraints:
+
+2 <= nums.length <= 104
+-109 <= nums[i] <= 109
+-109 <= target <= 109
+Only one valid answer exists.
+ 
+
+Follow-up: Can you come up with an algorithm that is less than O(n2) time complexity?
+
+*/
+
+const fs = require("fs");
+const arrayNums = fs.readFileSync("data.txt", "utf-8").split(",");
+let desiredTarget = arrayNums.pop();
+
+function findThePair(input, target) {
+    let object = {};
+
+    for (let i = 0; i < input.length; i++) {
+        if (Number.isInteger(object[target - input[i]])) {
+            return [object[target - input[i]], i];
+        }
+
+        object[input[i]] = i;
+    }
+
+    return "Sorry, couldn't find the pair...";

```

Comment:
I think the question originally asked to return [-1, -1] if you can't find a matching index.
In general, it's not ideal if a function returns different types, it can make it difficult to know what to expect from the function when you're using it.

Reply:
Strange, I actually originally had it return [-1,-1], but changed it in my final submission because I felt it stood out better.  I will change it back.
***BUT - can you cite your recollection that this is the required behaviour?***

Reply:
I reread the question and it actually says theres always a valid solution. I think I'm confusing this with the sorted version of this problem where we used two pointers. That one I think asked for `-1, -1`.

At any rate, I still do think returning two completely different types from a function can make that function difficult or confusing to use.